### PR TITLE
Audit - fix wrong direct dependency and params restructure

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/jfrog-cli-core#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli-core/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----
- Fix the "Direct Dependency" column in the audit results.
- Restructure the AuditParams parameters.

### CODE REVIEW

File: xray/commands/audit/generic/generic.go [hunk 1]
```diff
@@ -108,13 +108,13 @@ func (auditCmd *GenericAuditCommand) Run() (err error) {
...
 		return err
 	}
 
-	if auditCmd.Progress != nil {
-		err = auditCmd.Progress.Quit()
+	if auditCmd.Progress() != nil {
+		err = auditCmd.Progress().Quit()
 		if err != nil {
 			return
 		}
```
Comment (eyalbe4) [inline xray/commands/audit/generic/generic.go:120]: Can be simplified to - 
```
		if err = auditCmd.Progress().Quit(); err != nil {
			return
		}
```

File: xray/commands/curation/audit.go [hunk 2]
```diff
@@ -169,13 +169,13 @@ func (ca *CurationAuditCommand) Run() (err error) {
 			err = errors.Join(err, e)
 		}
 	}
-	if ca.Progress != nil {
-		if e := ca.Progress.Quit(); e != nil {
+	if ca.Progress() != nil {
+		if e := ca.Progress().Quit(); e != nil {
 			err = errors.Join(err, e)
 		}
```
Comment (eyalbe4) [inline xray/commands/curation/audit.go:175]: `errors.Join` handles nil values very nicely, so tha above can be simplified to - 
```
err = errors.Join(err, ca.Progress().Quit())
```
The same goes for line 178.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: xray/commands/audit/generic/generic.go (hunk 1 above)
Comment (omerzi) [reply xray/commands/audit/generic/generic.go:120]: done

File: xray/commands/curation/audit.go (hunk 2 above)
Comment (omerzi) [reply xray/commands/curation/audit.go:175]: done

**Pull request merged**